operator/controllers/lifecycle/keptnappcreationrequest/controller.go

```diff
@@ -190,14 +202,27 @@ func (r *KeptnAppCreationRequestReconciler) updateKeptnApp(ctx context.Context,
 			updated = true
 		}
 	}
+	return updated
+}
 
-	if !updated {
-		return nil
-	}
-
-	keptnApp.Spec.Version = computeVersionFromWorkloads(workloads.Items)
+func (r *KeptnAppCreationRequestReconciler) cleanupWorkloads(workloads *lifecycle.KeptnWorkloadList, keptnApp *lifecycle.KeptnApp) bool {
+	updated := false
+	for index, appWorkload := range keptnApp.Spec.Workloads {
+		foundWorkload := false
+		for _, workload := range workloads.Items {
+			workloadName := strings.TrimPrefix(workload.Name, fmt.Sprintf("%s-", keptnApp.Name))
+			if appWorkload.Name == workloadName {
+				foundWorkload = true
+				break
+			}
+		}
 
-	return r.Update(ctx, keptnApp)
+		if !foundWorkload {
+			keptnApp.Spec.Workloads = append(keptnApp.Spec.Workloads[:index], keptnApp.Spec.Workloads[index+1:]...)
```

Comment:
I remember that somewhere I read that we should `nil` the object being deleted to facilitate GC

Reply:
I think this we should do when working with maps, but not necessarry with arrays

Reply:
this is a slice, not an array 🤔

Reply:
I found the post about the GC and why it is important to `nil`-ify elements: https://stackoverflow.com/questions/28432658/does-go-garbage-collect-parts-of-slices/28432812#28432812

TL;DR: slices are struct built on top of an array. If the element is not `nil` it won't be deleted

Reply:
thanks for the link - my understanding originally was that `append` will create a new array,  but that's only the case if the length of the result exceeds the length of the original array. I have modified the function to create a new array with all workloads to be kept